resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/KubernetesVolumeUtils.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.deploy.k8s
+
+import org.apache.spark.SparkConf
+import org.apache.spark.deploy.k8s.Config._
+
+private[spark] object KubernetesVolumeUtils {
+
+  /**
+   * Extract Spark volume configuration properties with a given name prefix.
+   *
+   * @param sparkConf Spark configuration
+   * @param prefix the given property name prefix
+   * @return a Map storing with volume name as key and spec as value
+   */
+  def parseVolumesWithPrefix(
+    sparkConf: SparkConf,
+    prefix: String): Iterable[KubernetesVolumeSpec] = {
+    val properties = sparkConf.getAllWithPrefix(prefix)
+
+    val propsByTypeName: Map[(String, String), Array[(String, String)]] =
+      properties.flatMap { case (k, v) =>
+        k.split('.').toList match {
+          case tpe :: name :: rest => Some(((tpe, name), (rest.mkString("."), v)))
+          case _ => None
+        }
+      }.groupBy(_._1).mapValues(_.map(_._2))
+
+    propsByTypeName.map { case ((tpe, name), props) =>
+      val propMap = props.toMap
+      val mountProps = getAllWithPrefix(propMap, s"$KUBERNETES_VOLUMES_MOUNT_KEY.")
+      val options = getAllWithPrefix(propMap, s"$KUBERNETES_VOLUMES_OPTIONS_KEY.")
+
+      KubernetesVolumeSpec(
+        volumeName = name,
+        volumeType = tpe,
+        mountPath = mountProps(KUBERNETES_VOLUMES_PATH_KEY),
```

Comment:
What if a volume config key is malformed, e.g., missing `path` or `readOnly`?

Reply:
Then either spark-submit or driver die early with an exception. We might have config verification step to fail even sooner. What do you think?

Reply:
Yes, that is good to have. Basically it's good to have a check on the submission client side to make sure the config keys are properly formatted and fail fast if they are invalid.

Reply:
@liyinan926 now option if configuration is malformed it'll fail during driver submit step.